integration_tests/pages/create-goal.ts

```diff
@@ -0,0 +1,57 @@
+class CreateGoal {
+  selectGoalAutocompleteOption = (text: string, option: number) => {
+    this.addGoalAutoCompletionText(text)
+    cy.get(`#goal-input-autocomplete__option--${option}`).click()
+  }
+
+  addGoalAutoCompletionText = (text: string) => {
+    cy.get('#goal-input-autocomplete').type(text)
+  }
+
+  createGoal = (goalType: string) => {
+    cy.get(`a[href='/create-goal/${goalType}']`).click()
+    cy.url().should('include', `/create-goal/${goalType}`)
+  }
+
+  selectOtherAreasOfNeedRadio = (value: string) => {
+    cy.get('input[name="other-area-of-need-radio"]').check(value)
+  }
+
+  selectStartWorkingRadio = (value: string) => {
+    cy.get('input[name="start-working-goal-radio"]').check(value)
+  }
+
+  selectOtherAreasOfNeed = (values: string[]) => {
+    values.forEach(value => {
+      cy.get('.govuk-checkboxes').first().contains(value).click()
+    })
+  }
+
+  selectAchievementDate = (value: string) => {
+    cy.get('.govuk-radios').last().contains(value).click()
+  }
+
+  selectAchievementDateSomethingElse = (value: string) => {
+    cy.get('.govuk-radios').last().contains('Something else').click()
+    cy.get('.hmpps-js-datepicker-input').type(value)
+  }
+
+  clickButton = (value: string) => {
+    cy.contains('button', value).click()
+  }
+
+  createCompleteGoal = (value: number) => {
+    this.createGoal('accommodation')
+    this.selectGoalAutocompleteOption('acc', value)
+    this.selectOtherAreasOfNeedRadio('no')
+    this.selectStartWorkingRadio('yes')
+    this.selectAchievementDate('In 6 months')
+    this.clickButton('Save without steps')
+  }
+
+  selectFutureGoalsSubNavigation = () => {
+    cy.get('.moj-sub-navigation__list > li:nth-child(2)').click()
```

Comment:
I think we'd be better adding some addressable IDs to the elements on the relevant page and getting it directly. e.g currentGoals and futureGoals or something

Reply:
Those id would only be used for targeting by the test suite, and not the app. Not a fan of adding ids unused by the app itself.

Reply:
My understanding is that it's the recommended approach by most QA teams using browser automation tools, since it provides a layer of isolation between the specific page layout (e.g. we add a new "overdue" tab between "current" and "future") and element that we want to interact with. In fact most recommendations I've seen have been to add a `data-qa` or `data-testautomation` attribute, but I'm really not a fan of that unless IDs are auto generated by a tool that is creating the markup for you on the fly.

My view is that an ID only used for testing is OK compared to a potentially fragile test, but I'll leave it up to you.

Reply:
I'm not a fan of using more id's than necessary, as they can be targeted using anchors in the browser. https://docs.cypress.io/guides/references/best-practices#Selecting-Elements does suggest doing it with additional attributes for testing.

Reply:
Amended.